src/syntax/syntax.rkt

```diff
@@ -71,53 +80,81 @@
+                                             0.0))))])
+    (*loaded-ops* (cons 'operator (*loaded-ops*)))
     (table-set! operators 'operator
-                (make-hash (list (cons 'type type) (cons 'args args) (cons 'key value) ...)))))
+                (make-hash (append (list (cons 'type type) (cons 'args args) (cons 'key value) ...) nonffi)))))
 
 (define-operator (+ real real) real 
   [args '(2)] [type (hash 2 '(((real real) real) ((complex complex) complex)))]
   [fl +] [bf exact+] [cost 40]
   [->c/double (curry format "~a + ~a")]
   [->c/mpfr (curry format "mpfr_add(~a, ~a, ~a, MPFR_RNDN)")]
-  [->tex (curry format "~a + ~a")])
+  [->tex (curry format "~a + ~a")]
+  [nonffi +])
 
 (define-operator (- real [real]) real
   ;; Override the normal argument handling because - can be unary
   [args '(1 2)] [type (hash 1 '(((real) real) ((complex) complex)) 2 '(((real real) real) ((complex complex) complex)))]
   [fl -] [bf exact-] [cost 40]
   [->c/double (λ (x [y #f]) (if y (format "~a - ~a" x y) (format "-~a" x)))]
   [->c/mpfr (λ (out x [y #f]) (if y (format "mpfr_sub(~a, ~a, ~a, MPFR_RNDN)" out x y) (format "mpfr_neg(~a, ~a, MPFR_RNDN)" out x)))]
-  [->tex (λ (x [y #f]) (if y (format "~a - ~a" x y) (format "-~a" x)))])
+  [->tex (λ (x [y #f]) (if y (format "~a - ~a" x y) (format "-~a" x)))]
+  [nonffi -])
 
 (define-operator (* real real) real
   [args '(2)] [type (hash 2 '(((real real) real) ((complex complex) complex)))]
   [fl *] [bf exact*] [cost 40]
   [->c/double (curry format "~a * ~a")]
   [->c/mpfr (curry format "mpfr_mul(~a, ~a, ~a, MPFR_RNDN)")]
-  [->tex (curry format "~a \\cdot ~a")])
+  [->tex (curry format "~a \\cdot ~a")]
+  [nonffi *])
 
 (define-operator (/ real real) real
   [args '(2)] [type (hash 2 '(((real real) real) ((complex complex) complex)))]
   [fl /] [bf exact/] [cost 40]
   [->c/double (curry format "~a / ~a")]
   [->c/mpfr (curry format "mpfr_div(~a, ~a, ~a, MPFR_RNDN)")]
-  [->tex (curry format "\\frac{~a}{~a}")])
+  [->tex (curry format "\\frac{~a}{~a}")]
+  [nonffi /])
 
 ; Use C ffi to get numerical ops from libm
 (require ffi/unsafe ffi/unsafe/define)
-(define-ffi-definer define-libm #f
-  #:default-make-fail make-not-available)
+(define-ffi-definer define-libm #f)
 
 (define-syntax (define-operator/libm stx)
   (syntax-case stx (real libm)
     [(_ (operator real ...) real [libm id_d id_f] [key value] ...)
      (let ([num-args (length (cdr (syntax-e (cadr (syntax-e stx)))))])
        #`(begin
-           (define-libm id_d (_fun #,@(build-list num-args (λ (_) #'_double)) -> _double))
-           (define-libm id_f (_fun #,@(build-list num-args (λ (_) #'_float)) -> _float))
+           (define-libm id_d (_fun #,@(build-list num-args (λ (_) #'_double)) -> _double)
+                        #:fail (lambda ()
+                                 (*unknown-d-ops* (cons 'operator (*unknown-d-ops*)))
+                                 (λ args ((operator-info 'operator 'nonffi) args))))
```

Comment:
Use `apply` here so you don't need to constantly unpack the list.

    (apply (operator-info 'operator 'nonffi) args)

Reply:
`(operator-info 'operator 'nonffi)` is, per your earlier type annotation, a function from any number values to values. Yet you are not calling it here with values, you are calling it with a list of values. Instead of calling the function like `(lambda args (f args))` call it like this `(lambda args (apply f args))`. That way if you call the lambda `(g 1 2)` it will execute `(f 1 2)`. Your code instead execute `(f '(1 2))`.

Reply:
This is talking about the innermost lambda, the one that reads `(λ args ((operator-info 'operator 'nonffi) args)))`

Reply:
Ah, I see. Thanks for the clarification